### PR TITLE
[STRAT-107] Add initial script for weather API

### PR DESCRIPTION
**Pull request by AnhMai-bit**
- Set up code for energy calculation based on weather forecast API.
- The code is based on Solar team's code.
- Use One-call and Forecast for multiple days API from Openweather.
- Still in progress of editing.
- Goal: optimized model to obtain certain data from the API.

### CODE REVIEW

File: routemodel/data_retrieval/get_weather.py
Comment (e-wai) [inline routemodel/data_retrieval/get_weather.py:9]: Can we use relative file paths so that this works everywhere?

For eg. `path = os.path.join(os.path.dirname(__file__),'..','file')`

File: routemodel/data_retrieval/get_weather.py
Comment (e-wai) [inline routemodel/data_retrieval/get_weather.py:21]: We'll need to parse this response somehow, do you know how to?

**Review by e-wai — CHANGES_REQUESTED** [review]

**Review by ufshk — COMMENTED** [review]
The script seems to be good.

I'm just confused as to why this is in routemodel? @e-wai is it meant to be here?

**Comment by e-wai** [comment]
This looks very similar to https://github.com/uw-midsun/strategy/pull/116? Was there something I should be looking at

File: routemodel/data_retrieval/get_weather.py
Comment (e-wai) [inline routemodel/data_retrieval/get_weather.py:36]: I don't think this is a CSV file? `pd.read_csv` reads a CSV file into a pandas dataframe object

**Review by e-wai — CHANGES_REQUESTED** [review]

File: routemodel/data_retrieval/get_weather.py
Comment (AnhMai-bit) [reply routemodel/data_retrieval/get_weather.py]: I added some more at the bottom. Can you take a look?

File: routemodel/data_retrieval/get_weather.py
Comment (e-wai) [inline routemodel/data_retrieval/get_weather.py:47]: I think this might cause an error - because it's not in the same indentation level as the statement in line 42, the `data` variable is out of scope

File: routemodel/data_retrieval/get_weather.py
Comment (e-wai) [inline routemodel/data_retrieval/get_weather.py:42]: Why the `rb` option? Also, I don't think this file exists yet - it won't be possible to read from it?

**Review by e-wai — CHANGES_REQUESTED** [review]
Hey - can we do a quick call sometime? My review is a bit lacking because I'm not entirely sure what steps you're trying to accomplish in the last few lines?

**Review by e-wai — CHANGES_REQUESTED** [review]
So the bottom section looks good now, but we should build this on top of the code that was merged in https://github.com/uw-midsun/strategy/pull/116. We might also be interested in fields other than current temperature -> wind speed, wind direction, precipitation. Let me know if you want to set up a pairing session to look through this + the Git stuff to set it up

Side note - we should also probably update documentation for the `routemodel` directory

**Comment by ufshk** [comment]
> 
> 
> So the bottom section looks good now, but we should build this on top of the code that was merged in #116. We might also be interested in fields other than current temperature -> wind speed, wind direction, precipitation. Let me know if you want to set up a pairing session to look through this + the Git stuff to set it up
> 
> Side note - we should also probably update documentation for the `routemodel` directory

Wait, why is this going into `routemodel` at all?

**Comment by e-wai** [comment]
> > So the bottom section looks good now, but we should build this on top of the code that was merged in #116. We might also be interested in fields other than current temperature -> wind speed, wind direction, precipitation. Let me know if you want to set up a pairing session to look through this + the Git stuff to set it up
> > Side note - we should also probably update documentation for the `routemodel` directory
> 
> Wait, why is this going into `routemodel` at all?

Weather data is part of the data retrieval for info on the route? It seems to fit in that category - or do you mean since we are updating this information during the race, which we aren't doing with the rest of routemodel?

**Review by e-wai — APPROVED** [review]
This looks good to me, but someone else should review before this gets merged?

**Review by ufshk — DISMISSED** [review]
Think I'd want the file name to change and for logic to be encapsulated in functions.

File: routemodel/data_retrieval/get_weather.py
Comment (e-wai) [inline routemodel/data_retrieval/get_weather.py:10]: I think this can go before/outside of the function

File: routemodel/data_retrieval/get_weather.py
Comment (e-wai) [inline routemodel/data_retrieval/get_weather.py:13]: I would encourage you to add this as a parameter to the function so it can be passed/easily changed

File: routemodel/data_retrieval/get_weather.py
Comment (e-wai) [inline routemodel/data_retrieval/get_weather.py:40]: I think this deserves its own function? @ufshk thoughts? It has a pretty distinct function

**Review by e-wai — CHANGES_REQUESTED** [review]

**Review by e-wai — APPROVED** [review]

**Pull request merged**
